src/apps/lwaftr/fragmentv4.lua

```diff
@@ -0,0 +1,105 @@
+module(..., package.seeall)
+
+local constants = require("apps.lwaftr.constants")
+local get_ihl = require("apps.lwaftr").get_ihl
+local packet = require("core.packet")
+local bit = require("bit")
+local ffi = require("ffi")
+
+local cast = ffi.cast
+local C = ffi.C
+local band, bor = bit.band, bit.bor
+local ceil = math.ceil
+
+local int16_ptr_t = ffi.typeof("uint16_t*")
+
+local total_length_offset = constants.ethernet_header_size + constants.o_ipv4_total_length
+local frag_id_offset = constants.ethernet_header_size + constants.o_ipv4_identification
+local flags_and_frag_offset_offset = constants.ethernet_header_size + constants.o_ipv4_flags
+local ihl_and_ver_offset = constants.ethernet_header_size + constants.o_ipv4_ver_and_ihl
+
+-- Constants to manipulate the flags next to the frag-offset field directly
+-- as a 16-bit integer, without needing to shift the 3 flag bits.
+local flag_dont_fragment_mask  = 0x4000
+local flag_more_fragments_mask = 0x2000
+local frag_offset_field_mask   = 0x1FFF
+
+
+local function get_u16_at_offset(pkt, offset)
+   return C.ntohs(cast(int16_ptr_t, pkt.data + offset)[0])
+end
+
+local function set_u16_at_offset(pkt, offset, value)
+   cast(int16_ptr_t, pkt.data + offset)[0] = C.htons(value)
+end
+
+-- TODO: Consider security/performance tradeoffs of randomization
+local fresh_frag_id = (function ()
+   local internal_frag_id = 0x4242
+   return function ()
+      internal_frag_id = (internal_frag_id + 1) % 0xFFFF
```

Comment:
Forget about this one. Modulo 0xFFFF will only give values in the 0-0xFFFE range, 65535 values, and not 65536 values, which is what it was intended.

In addition to that, calculating the reminder of a division with bitwise and arithmetic is usually faster than using mod (I have no numbers for LuaJIT, but is a well-known optimization). So, in conclusion, I think the code is fine as it originally was. Sorry for the noise.